javascript/ql/lib/semmle/javascript/frameworks/Spife.qll

```diff
@@ -218,29 +208,25 @@ module Spife {
    * A Spife response source, that is, the response variable used by a
    * route handler.
    */
-  private class ReplySource extends Http::Servers::ResponseSource instanceof DataFlow::CallNode {
-    ReplySource() {
-      // const reply = require("@npm/spife/reply")
+  private class ReplyCall extends API::CallNode {
+    ReplyCall() {
       // reply(resp)
+      this = API::moduleImport(["@npm/spife/reply", "spife/reply"]).getACall()
+      or
       // reply.header(resp, 'foo', 'bar')
-      this = API::moduleImport(["@npm/spife/reply", "spife/reply"]).getACall() or
       this = API::moduleImport(["@npm/spife/reply", "spife/reply"]).getAMember().getACall()
     }
 
-    private DataFlow::SourceNode reachesHandlerReturn(DataFlow::TypeTracker t) {
-      result = this and
-      t.start()
-      or
-      exists(DataFlow::TypeTracker t2 | result = this.reachesHandlerReturn(t2).track(t2, t))
+    predicate isDirectReplyCall() {
+      this = API::moduleImport(["@npm/spife/reply", "spife/reply"]).getACall()
```

Comment:
This is duplication from the charpred, you could save whether it's a direct call as a field?